src/hooks/useBridgedBlocks.ts

```diff
@@ -39,34 +39,26 @@ const useBridgedBlocks = ({ isApiReady, api, chain }: Props) => {
   const [bestBridgedFinalizedBlock, setBestBridgedFinalizedBlock] = useState('');
 
   const { bridgedGrandpaChain } = getSubstrateDynamicNames(chain);
-  useEffect(() => {
-    if (!api || !isApiReady || !chain) {
-      return;
-    }
+  useEffect((): (() => void) => {
+    const shouldProceed: boolean = !!(api && isApiReady && chain);
 
-    let unsubBestFinalized: () => void;
-    let unsubImportedHeaders: () => void;
+    let unsubImportedHeaders: Promise<VoidFn> | null;
+    const unsubBestFinalized: Promise<VoidFn> | null = shouldProceed
+      ? api.query[bridgedGrandpaChain].bestFinalized((res: CodecHeaderId) => {
+          const bestBridgedFinalizedBlock = res.toString();
+          unsubImportedHeaders = shouldProceed
```

Comment:
Checking `shouldProceed` doesn't make any sense here. It is guaranteed to be `true`, cause otherwise we would never run `bestFinalized` earlier, ergo we wouldn't end up in this block. I guess you meant to re-evaluate here, but it still doesn't make that much sense, cause the values will be exactly the same anyway.

Maybe just early exit the entire closure right at the beginning:
```js
 useEffect((): (() => void) => {
    const shouldProceed: boolean = !!(api && isApiReady && chain);
    if (!shouldProceed) return;
    // do the subscription stuff here
    // ...
});